### ISSUE TITLE
[6.0.0/7.0.0] built-in types not caught when importing from package main

### ISSUE THREAD

**Issue by AviVahl** [puppeteer/puppeteer#6807]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 6.0.0
* Platform / OS version: Fedora 33
* Node.js version: 14.15.4

**What steps will reproduce the problem?**

install puppteer in a typescript project and import it

_Please include code that reproduces the issue._

1. `import puppeteer from 'puppeteer';`

**What is the expected result?**

Built-in types should be caught. Changelog says "Puppeteer now ships with built-in TypeScript definitions. This means you no longer need to install @types/puppeteer."

**What happens instead?**

Types aren't picked up by typescript. `package.json` points to `cjs-entry.js`, which doesn't have a .d.ts next to it re-exporting `./lib/cjs/puppeteer/node`.

**Comment by AviVahl** [comment]
Oh, and because the original code is esm with a default export, and `cjs-entry.js` unwraps it, the .d.ts should probably look like: 
```ts
import Puppeteer from "./lib/cjs/puppeteer/node";
export = Puppeteer;
```
and `puppeteer/node.ts` will need a namespace called Puppeteer which re-exports types you want on that namespace.

**Comment by AviVahl** [comment]
This issue is fixed in 7.0.1.

Now that the `@types` are caught, everything is breaking because they differ from `@types/puppeteer`. :(
Will report issues.
